src/lib/disk_cache/lmdb/disk_cache.ml

```diff
@@ -47,17 +47,24 @@ module Make (Data : Binable.S) = struct
     Rw.get ~env db idx |> Option.value_exn
 
   let put ({ env; db; counter; logger; garbage } : t) (x : Data.t) : id =
-    (* TODO: we may reuse IDs by pulling them from the `garbage` hash set *)
-    let idx = !counter in
-    incr counter ;
+    let idx =
+      match Hash_set.find garbage ~f:(const true) with
+      | None ->
+          incr counter ; !counter - 1
+      | Some reused_id ->
+          [%log debug] "Reusing LMDB key %d for a new KV pair" reused_id
+            ~metadata:[ ("index", `Int reused_id) ] ;
+          Hash_set.remove garbage reused_id ;
```

Comment:
```suggestion
```

in some code below we'll overwrite the value anyway:

```ocaml
    Rw.set ~env db idx x ;
```

Reply:
Yes, I think you're right @glyh. This is `Hash_set.remove` here and not `Rw.remove` - the reused ID needs to be removed from the hash set tracking if we're going to reuse it.